platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultGradleEnterprisePluginAdapter.java

```diff
@@ -95,14 +103,28 @@ public void buildFinished(@Nullable Throwable buildFailure) {
                 public Throwable getFailure() {
                     return buildFailure;
                 }
+
+                @Override
+                public Collection<Problem> getProblems() {
```

Comment:
🤔 It would be very good to be able to obtain the problems for each cause of the failure individually. That is something I can build on top of the infrastructure you provide here.